### PR TITLE
NOMAD 4.0.1

### PR DESCRIPTION
**Pull request by amontoison**

### CODE REVIEW

**Comment by giordano** [comment]
Apparently, we need to build for
```julia
expand_cxxstring_abis(supported_platforms(); skip=Sys.isapple)
```
(not sure why it complains about FreeBSD but not for macOS, since we use GCC in both cases)

File: N/NOMAD/build_tarballs.jl [hunk 1]
```diff
@@ -1,11 +1,11 @@
 using BinaryBuilder
```
Comment (giordano) [inline N/NOMAD/build_tarballs.jl:1]:
```suggestion
using BinaryBuilder, Pkg
```

**Comment by amontoison** [comment]
I don't understand why we need `skip=Sys.isapple` but I trust you :)

**Comment by giordano** [comment]
Without keyword arguments, `expand_cxxstring_abis` expands only the platforms which use GCC by default (because the C++ string ABI is a [GCC thing](https://gcc.gnu.org/onlinedocs/libstdc++/manual/using_dual_abi.html)), that are all but macOS and FreeBSD (or `Sys.isbsd()`).  However here you're building with GCC everywhere, so auditor finds the cxxstring tags also when building for FreeBSD.  No idea why it doesn't find it for macOS (I didn't check the library)

**Comment by amontoison** [comment]
Thanks @giordano ! It could solve our problems on FreeBSD, NOMAD.jl was not working on FreeBSD when I testes with Cirrus CI.

**Comment by giordano** [comment]
That's interesting, but I wouldn't expect so, because GCC 8 would use the C++11 string ABI, which is what your Julia is using anyway.  You'd have problems when using a libstdc++ with only C++03 string ABI, which is quite rare (but we want to support anyway)

**Comment by amontoison** [comment]
I had this error when I tested the interface with FreeBSD for information : https://github.com/bbopt/NOMAD.jl/runs/874036585
I will activate Cirrus CI again to test with this new version of NOMAD_jll.

**Comment by giordano** [comment]
The offending function is
```console
% echo "_ZN11NOMAD_4_0_010Parameters17setSpValueDefaultImEEvRKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEET_" | c++filt 
void NOMAD_4_0_0::Parameters::setSpValueDefault<unsigned long>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned long)
```
which does have a reference to the C++ string ABI, but I would expect Julia's libstdc++ and the library's API to match in this case

**Pull request merged**